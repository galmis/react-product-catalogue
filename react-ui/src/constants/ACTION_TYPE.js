// @flow

const ACTION_TYPE: Object = Object.freeze({
  FILTER: 'FILTER'
});

export default ACTION_TYPE;
