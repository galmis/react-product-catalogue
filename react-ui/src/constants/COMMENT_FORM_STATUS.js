// @flow

import { SUCCESS, WARNING, ERROR } from './COMMENT_FORM_STATUS_TYPE';

const COMMENT_FORM_STATUS = Object.freeze({
  ['201']: {
    type: SUCCESS,
    code: 201,
    message: 'Komentaras išsiųstas sėkmingai.',
  },
  ['400']: {
    type: ERROR,
    code: 400,
    message: 'Komentaras neišsiųstas. Palaukite 15 sekundžių ir bandykite dar kartą.'
  },
  ['409']: {
    type: ERROR,
    code: 409,
    message: 'Komentaras neišsiųstas. Jūsų komentaras su tokiu pačiu tekstu jau egzistuoja.'
  }
});

export default COMMENT_FORM_STATUS;
