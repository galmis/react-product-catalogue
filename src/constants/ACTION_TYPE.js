// @flow

const ACTION_TYPE: Object = Object.freeze({

});

export default ACTION_TYPE;
