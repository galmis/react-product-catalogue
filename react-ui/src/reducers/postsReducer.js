// @flow

import {
  FETCH_POSTS_SUCCESS,
  FETCH_POSTS
} from '../constants/ACTION_TYPE';
import { POSTS } from '../constants/RESOURCE_REF';
import type { Action, NormalizedData } from '../types';

const initialState: Object = {
  postsById: {},
  totalRecords: 0,
  totalPages: 0,
  selectedPage: 1,
  fetchedPages: {},
  isLoading: false
};

export {
  initialState
}

export default function postsReducer(state: Object = initialState, action: Action) {

  switch(action.type) {
    case FETCH_POSTS_SUCCESS: {
      const selectedPage = action.payload.selectedPage;
      const fetchedPages = { ...state.fetchedPages };
      fetchedPages[selectedPage] = action.payload.data.result;
      const newState =  {
        fetchedPages,
        selectedPage,
        isLoading: false,
        totalRecords: action.payload.totalRecords,
        totalPages: action.payload.totalPages,
        postsById: {
          ...state.postsById,
          ...action.payload.data.entities.dataById
        }
      };

      return newState;
    }
    case FETCH_POSTS: {
      return {
        ...state,
        isLoading: true
      }
    }
  }

  return state;
}
