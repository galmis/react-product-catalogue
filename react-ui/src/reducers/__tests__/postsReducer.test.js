// @flow
'use strict';

jest.dontMock('../postsReducer.js');

import postsReducer from '../postsReducer';
import deepFreeze from 'deep-freeze';

import { FETCH_POSTS_SUCCESS } from '../../constants/ACTION_TYPE';

describe('postsReducer', () => {

  it('FETCH_POSTS_SUCCESS, initialState', () => {
    const action = {
      type: FETCH_POSTS_SUCCESS,
      payload: {
        data: {
          result: ['1'],
          entities: {
            dataById: {
              '1': 'foo'
            }
          },
        },
        selectedPage: 2,
        totalRecords: 12,
        totalPages: 2
      }
    };
    deepFreeze(action);

    expect(postsReducer(undefined, action)).toMatchSnapshot();
  });


  it('FETCH_POSTS_SUCCESS, existing state', () => {
    const action = {
      type: FETCH_POSTS_SUCCESS,
      payload: {
        data: {
          result: ['3'],
          entities: {
            dataById: {
              '3': 'bar'
            }
          },
        },
        selectedPage: 2,
        totalRecords: 8,
        totalPages: 1
      }
    };
    deepFreeze(action);


    const oldState = {
      postsById: {
        '1': 'foo',
        '2': 'xyz'
      },
      totalRecords: 12,
      totalPages: 2,
      selectedPage: 1,
      fetchedPages: {
        '1': ['1', '2']
      }
    }
    deepFreeze(oldState);
    expect(postsReducer(oldState, action)).toMatchSnapshot();
  });

});
