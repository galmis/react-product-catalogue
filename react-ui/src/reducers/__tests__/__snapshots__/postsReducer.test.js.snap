exports[`postsReducer FETCH_POSTS_SUCCESS, initialState 1`] = `
Object {
  "fetchedPages": Object {
    "2": Array [
      "1",
    ],
  },
  "isLoading": false,
  "postsById": Object {
    "1": "foo",
  },
  "selectedPage": 2,
  "totalPages": 2,
  "totalRecords": 12,
}
`;

exports[`postsReducer FETCH_POSTS_SUCCESS, new page 1`] = `
Object {
  "fetchedPages": Object {
    "1": Array [
      "1",
      "2",
    ],
    "2": Array [
      "3",
    ],
  },
  "isLoading": false,
  "postsById": Object {
    "1": "foo",
    "2": "xyz",
    "3": "bar",
  },
  "selectedPage": 2,
  "totalPages": 2,
  "totalRecords": 8,
}
`;

exports[`postsReducer FETCH_POSTS_SUCCESS, overwrite existing page 1`] = `
Object {
  "fetchedPages": Object {
    "1": Array [
      "2",
      "3",
    ],
  },
  "isLoading": false,
  "postsById": Object {
    "1": "foo",
    "2": "alio",
    "3": "bar",
  },
  "selectedPage": 1,
  "totalPages": 1,
  "totalRecords": 8,
}
`;
